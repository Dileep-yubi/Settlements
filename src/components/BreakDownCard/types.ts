import { Header } from "components/EmiPlanComponent/EmiPlanComponent";
import { Payment } from "../BreakDown/types";

export type StateProps = {
  payments: Array<Payment>;
};

export type OwnProps = {
  plan: Array<Payment>;
  header: Header;
};

export type BreakDownCardProps = StateProps & OwnProps;
