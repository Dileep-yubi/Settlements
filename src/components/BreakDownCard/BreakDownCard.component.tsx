import { EmiPlanHeader } from "components/EmiComponent/EmiPlanHeader";
import styled from "styled-components";
import BreakDown from "../BreakDown/BreakDown.component";
import "./BreakDownCard.css";
import { BreakDownCardProps } from "./types";
export const Divider = styled.div`
  width: 100%;
  height: 1px;
  border-top: 1px solid #d6d3d3;
`;

export const BreakDownCard: React.FC<BreakDownCardProps> = ({
  payments,
  plan,
  header,
}) => {
  return (
    <div className="emiComponent emiComponentSelected">
      <div style={{ padding: "15px 15px 0px 15px", gap: "3em" }}>
        <EmiPlanHeader title={header} isSelected={false} />
        <BreakDown summary={plan} />
      </div>
    </div>
  );
};
