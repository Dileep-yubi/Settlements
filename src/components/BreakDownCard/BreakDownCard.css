.emiComponent {
  border: 1px solid #d6d3d3;
  border-radius: 12px;
  background: #f8f8fbe1;
  width: 100%;
  min-width: fit-content;
  height: fit-content;
}

.emiComponentSelected {
  border: 1px solid #0e0778;
}
