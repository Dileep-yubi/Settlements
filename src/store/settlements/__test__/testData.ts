import { Payment } from "../../../components/BreakDown/types";

export const payments: Array<Payment> = [
  {
    title: "Tenure",
    period: '9 months'
  },
  {
    title: "EMI",
    value: 3339,
    currency: "USD",
  },
  { title: "R.O.I", interest: "1%" },
  {
    title: "Interest",
    value: 150,
    currency: "USD",
  },
  {
    title: "Total amount payable",
    value: 1000,
    currency: "USD",
  },
];
