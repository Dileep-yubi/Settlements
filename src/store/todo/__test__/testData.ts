import { Payment } from "../../../components/BreakDown/types";

export const payments: Array<Payment> = [
  {
    title: "Tenure",
    value: "9 months",
  },
  {
    title: "EMI",
    value: "3339",
    currency: "USD",
  },
  { title: "R.O.I", value: "1%" },
  {
    title: "Interest",
    value: "150",
    currency: "USD",
  },
  {
    title: "Total amount payable",
    value: "1000",
    currency: "USD",
  },
];
