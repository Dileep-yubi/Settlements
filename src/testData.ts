import { Plan } from "components/EmiPlanComponent/EmiPlanComponent";

export const headerComponent: Array<Plan> = [
  {
    header: {
      currency: "USD",
      value: 3339,
      period: "3 months",
      interest: "1%",
    },
    children: [
      {
        title: "Tenure",
        period: "4 months",
      },
      {
        title: "Interest",
        value: 25,
        currency: "USD",
      },
      {
        title: "Total amount payable",
        value: 435,
        currency: "USD",
      },
    ],
  },
  {
    header: {
      currency: "USD",
      value: 1839,
      period: "6 months",
      interest: "1.5%",
    },
    children: [
      {
        title: "Tenure",
        period: "6 months",
      },
      {
        title: "Interest",
        currency: "USD",
        value: 75,
      },
      {
        title: "Total amount payable",
        value: 10075,
        currency: "USD",
      },
    ],
  },
  {
    header: {
      currency: "USD",
      value: 1277,
      period: "9 months",
      interest: "2%",
    },
    children: [
      {
        title: "Tenure",
        period: "9 months",
      },
      {
        title: "Interest",
        value: 150,
        currency: "USD",
      },
      {
        title: "Total amount payable",
        value: 10150,
        currency: "USD",
      },
    ],
  },
];
