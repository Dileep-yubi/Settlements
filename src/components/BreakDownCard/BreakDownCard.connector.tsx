import { connect } from "react-redux";
import { AppState } from "../../store";
import { BreakDownCard } from "./BreakDownCard.component";
import { OwnProps, StateProps } from "./types";

const mapStateToProps = ({ todoState }: AppState): StateProps => ({
  payments: todoState.payments,
});

export default connect<StateProps, null, OwnProps, any>(
  mapStateToProps,
  null
)(BreakDownCard);
