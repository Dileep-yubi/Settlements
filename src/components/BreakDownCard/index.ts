import BreakDownCard from "./BreakDownCard.connector";
export default BreakDownCard;
