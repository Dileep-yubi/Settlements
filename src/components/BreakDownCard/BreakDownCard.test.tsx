import { render, screen } from "@testing-library/react";
import { Provider } from "react-redux";
import { payments } from "../../store/settlements/__test__/testData";
import BreakDownCard from "./";
import reduxMockStore from "redux-mock-store";
import { headerComponent } from "testData";

describe("BreakDownCard", () => {
  const mockStore = reduxMockStore(),
    store = mockStore({
      todoState: { payments: payments },
    });
  const renderComponent = () =>
    render(
      <Provider store={store}>
        <BreakDownCard plan={payments} header={headerComponent[0].header} />
      </Provider>
    );
  it("snapshot", () => {
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
    expect(screen.getByText("EMI")).toBeInTheDocument();
    expect(screen.getByText("USD 1000")).toBeInTheDocument();
  });
});
