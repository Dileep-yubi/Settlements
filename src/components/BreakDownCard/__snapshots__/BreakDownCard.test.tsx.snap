// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreakDownCard snapshot 1`] = `
<div>
  <div
    class="emiComponent emiComponentSelected"
  >
    <div
      style="padding: 15px 15px 0px 15px; gap: 3em;"
    >
      <div
        class="sc-dkrFOg iMlCA-d"
      >
        <span>
          USD 3339 for 3 months @1% p.a
        </span>
      </div>
      <div
        style="font-size: 14px; font-weight: normal;"
      >
        <div
          class="sc-jSUZER dGOTlI"
          style="display: flex; justify-content: space-between;"
        >
          <span>
            Tenure
          </span>
          <span>
            9 months
          </span>
        </div>
      </div>
      <div
        style="font-size: 14px; font-weight: normal;"
      >
        <div
          class="sc-gKPRtg hKTseF"
          data-testid="divider"
        />
        <div
          class="sc-jSUZER dGOTlI"
          style="display: flex; justify-content: space-between;"
        >
          <span>
            EMI
          </span>
          <span>
            USD 3339
          </span>
        </div>
      </div>
      <div
        style="font-size: 14px; font-weight: normal;"
      >
        <div
          class="sc-gKPRtg hKTseF"
          data-testid="divider"
        />
        <div
          class="sc-jSUZER dGOTlI"
          style="display: flex; justify-content: space-between;"
        >
          <span>
            R.O.I
          </span>
          <span>
            1%
          </span>
        </div>
      </div>
      <div
        style="font-size: 14px; font-weight: normal;"
      >
        <div
          class="sc-gKPRtg hKTseF"
          data-testid="divider"
        />
        <div
          class="sc-jSUZER dGOTlI"
          style="display: flex; justify-content: space-between;"
        >
          <span>
            Interest
          </span>
          <span>
            USD 150
          </span>
        </div>
      </div>
      <div
        style="font-size: 14px; font-weight: normal;"
      >
        <div
          class="sc-gKPRtg hKTseF"
          data-testid="divider"
        />
        <div
          class="sc-jSUZER dGOTlI"
          style="display: flex; justify-content: space-between;"
        >
          <span>
            Total amount payable
          </span>
          <span>
            USD 1000
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
